### PR TITLE
Fix Building.user_requested_exports under wasm backend

### PR DESCRIPTION
**Pull request by sbc100**
This list should contain names with the '_' prefix. With this the
metadce process was removing them.

Fixes: #8194

### CODE REVIEW

**Comment by sbc100** [comment]
This was a little more complicated that I thought.  Should be good now though PTAL.

wasm3.test_llvm_used now passes

File: tests/test_other.py [hunk 1]
```diff
@@ -7943,9 +7943,9 @@ def run(*args):
       run(['-O1'],  9, [], ['waka'],  8095,  2, 12, 10) # noqa
       run(['-O2'],  9, [], ['waka'],  8077,  2, 12, 10) # noqa
       # in -O3, -Os and -Oz we metadce, and they shrink it down to the minimal output we want
-      run(['-O3'],  0, [], [],          61,  0,  1,  1) # noqa
-      run(['-Os'],  0, [], [],          61,  0,  1,  1) # noqa
-      run(['-Oz'],  0, [], [],           8,  0,  0,  0) # noqa
+      run(['-O3'],  0, [], [],          85,  0,  2,  2) # noqa
+      run(['-Os'],  0, [], [],          85,  0,  2,  2) # noqa
+      run(['-Oz'],  0, [], [],          54,  0,  1,  1) # noqa
```
Comment (kripken) [inline tests/test_other.py]: I think the code is right, but the results are slightly surprising.

Was `-Oz` was basically empty and incorrect before, I suppose? And what is the cause of the smaller increase in the others?

Comment (sbc100) [reply tests/test_other.py]: Its looks like its due to the __post_instantiate function being includes. And also the name section is still in there... which is a separate bug that we should look at.  But this change is strictly more correct.

Comment (kripken) [reply tests/test_other.py]: I agree this change is more correct, yeah. Would be good to improve things so post-instantiate etc. can also be metadce'd out, so we are comparable to the asm2wasm path (55 bytes vs 85, in -Os/-Oz).

**Pull request merged**